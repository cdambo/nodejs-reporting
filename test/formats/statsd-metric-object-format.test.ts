import StatsDMetricObjectFormat from "../../src/formats/statsd-metric-object-format";

describe("StatsDMetricObjectFormat", (): void => {
  it("Formats the metric to an object", (): void => {
    expect(
      StatsDMetricObjectFormat({
        stat: "metric_name",
        a: 1,
        b: 2
      })
    ).toMatchSnapshot();
  });

  it("Formats the tags to snake case keys", (): void => {
    expect(
      StatsDMetricObjectFormat({
        stat: "metric_name",
        a: 1,
        b: 2,
        tags: {
          camelCaseTag1: "tagOne",
          camelCaseTag2: "tag_two",
          anotherTag: 5
        }
      })
    ).toMatchSnapshot();
  });

  it("Removes leading non-letters from metric name", (): void => {
    expect(
      StatsDMetricObjectFormat({ stat: "123._-/*()#$@!a_real_metric" })
    ).toMatchSnapshot();
  });

  it("Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name", (): void => {
    expect(
      StatsDMetricObjectFormat({
        stat: "a-real_Metric.Name_with+bad#chars!@-#$%^&*()9"
      })
    ).toMatchSnapshot();
  });
});
