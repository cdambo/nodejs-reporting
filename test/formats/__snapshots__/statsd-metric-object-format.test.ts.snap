// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsDMetricObjectFormat Formats the metric to an object 1`] = `
Object {
  "a": 1,
  "b": 2,
  "stat": "metric_name",
  "tags": Object {},
}
`;

exports[`StatsDMetricObjectFormat Formats the tags to snake case keys 1`] = `
Object {
  "a": 1,
  "b": 2,
  "stat": "metric_name",
  "tags": Object {
    "another_tag": 5,
    "camel_case_tag_1": "tagOne",
    "camel_case_tag_2": "tag_two",
  },
}
`;

exports[`StatsDMetricObjectFormat Removes leading non-letters from metric name 1`] = `
Object {
  "stat": "a_real_metric",
  "tags": Object {},
}
`;

exports[`StatsDMetricObjectFormat Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name 1`] = `
Object {
  "stat": "a_real_Metric.Name_with_bad_chars___________9",
  "tags": Object {},
}
`;
